src/wp-includes/default-filters.php

```diff
@@ -568,6 +568,7 @@
 add_action( 'enqueue_block_editor_assets', 'wp_enqueue_editor_format_library_assets' );
 add_action( 'enqueue_block_editor_assets', 'wp_enqueue_global_styles_css_custom_properties' );
 add_filter( 'wp_print_scripts', 'wp_just_in_time_script_localization' );
+add_action( 'wp_print_scripts', 'wp_print_delayed_inline_script_loader' );
```

Comment:
Have we tested that this includes the utility script correctly? I'm not at all saying it doesn't, just that it's worth double-checking, e.g.:
* Are we sure we've avoided a chance for the script getting printed twice? The `wp_print_scripts` action seems like it's called in WP core in a few places, so I wonder about that.
* Are we sure the script is printed early enough? We need to ensure it's printed before any script tags using its JS function in an `onload` attribute are printed.

Reply:
If we go the route of only using this function for `defer` scripts (since `async` scripts by nature have unpredictable load order and thus should not need to wait for them to load), it won't actually matter where we print the script. I suggest we change this to:

```suggestion
add_action( 'wp_print_footer_scripts', 'wp_print_delayed_inline_script_loader' );
```

The `load` event will only fire for loaded `defer` scripts after the DOM has fully loaded, and thus we will be guaranteed that the inline script defining `wpLoadAfterScripts` will have already evaluated.

This will also address the issue of the script being printed twice.

Reply:
Scratch my above comment. With https://github.com/10up/wordpress-develop/pull/62 it would be important for the script to be printed _before_ any other scripts are printed.

Now, in regards to the concern about the script being printed _twice_, I think that the `wp_print_scripts` action _is_ actually the better one. Why? Because of the call to the `has_delayed_inline_script` method in the `wp_print_delayed_inline_script_loader` function:

https://github.com/WordPress/wordpress-develop/blob/e91b02cd13efdfa62a48a4a7a60d158d2e2057c7/src/wp-includes/script-loader.php#L1854-L1857

It could very well be that an `async`/`defer` script is not enqueued at `wp_enqueue_scripts` (at `wp_head`) but it is later on in the execution of the page as a late script. In this case it will end up getting printed at `wp_print_footer_scripts`. So what I suggest is that we try printing the script at the `wp_print_scripts` action (so again at `wp_head` and `wp_footer`) but that we add logic so that the script is only output once. 

One possible way to achieve this is inside of the condition to add:

```php
remove_action( 'wp_print_scripts', 'wp_print_delayed_inline_script_loader' );
```

Otherwise there could be a flag set on the `WP_Scripts` class via some member variable. 

Or else a static variable could be used in the function, but that would be more difficult to test. Lastly a global variable could be used, but please no 😄

Reply:
Another option would be to add a check here before `wp_scripts()->has_delayed_inline_script()` to see if there are any handles in `wp_scripts()->done` which are `async`/`defer`. If so, we can presume that `wp_print_delayed_inline_script_loader()` has already printed its script.

Reply:
@westonruter I still need to consider and test your comment above RE script printing twice, however the aforementioned PR above (https://github.com/10up/wordpress-develop/pull/62) has now been merged into this feature base branch.

Reply:
I have indeed found there are problems here. I've opened https://github.com/10up/wordpress-develop/pull/75 to add test cases and address the issues.

Reply:
Ready for review: https://github.com/10up/wordpress-develop/pull/75